dss-api.yml

```diff
@@ -660,6 +660,258 @@ paths:
+          schema:
+            type: object
+            properties:
+              uuid:
+                description: A RFC4122-compliant ID for the collection.
+                type: string
+                pattern: "[A-Za-z0-9]{8}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{12}"
+            required:
+              - uuid
+        default:
+          description: Unexpected error
+          schema:
+            allOf:
+              - $ref: '#/definitions/Error'
+              - type: object
+                properties:
+                  code:
+                    type: string
+                    description: |
+                      Machine-readable error code.  The types of return values should not be changed lightly.
+
+                      The code `invalid_link` indicates that an item listed in the collection contents field of the
+                      input could not be resolved (either because the file, bundle, or collection does not exist on the
+                      replica specified, or because the JSON Pointer reference could not be resolved within the
+                      specified file).
+                    enum: [unhandled_exception, Forbidden, OAuthResponseProblem, not_found, invalid_link]
+                required:
+                  - code
+  /collections/{uuid}:
+    get:
+      summary: Retrieve a collection given a UUID.
+      description: |
+        Given a collection UUID, return the associated collection object.
+      security:
+        - googAuth:
+            - https://www.googleapis.com/auth/userinfo.email
+      parameters:
+        - name: uuid
+          in: path
+          description: A RFC4122-compliant ID for the collection.
+          required: true
+          type: string
+          pattern: "[A-Za-z0-9]{8}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{12}"
+        - name: replica
+          in: query
+          description: Replica to fetch from.
+          required: true
+          type: string
+          enum: [aws, gcp, azure]
+        - name: version
+          in: query
+          description: |
+            Timestamp of collection creation in RFC3339.  If this is not provided, the latest version is returned.
+          required: false
+          type: string
+          format: date-time
+      responses:
+        200:
+          description: OK
+          schema:
+            type: object
+            properties:
+              collection:
+                $ref: "#/definitions/Collection"
+        default:
+          description: Unexpected error
+          schema:
+            allOf:
+              - $ref: '#/definitions/Error'
+              - type: object
+                properties:
+                  code:
+                    type: string
+                    description: Machine-readable error code.  The types of return values should not be changed lightly.
+                    enum: [unhandled_exception, Forbidden, OAuthResponseProblem, not_found]
```

Comment:
out of the scope of this diff, but we should move not_found to a different 404 definition.

Reply:
I'm not sure if I follow, can you elaborate?

Reply:
Right now, all these error can be returned for any non-201 code.  we'd ideally only want it for not_found to be returned for 404, and 404 should only allow not_found.